mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/annotation/XmlHintsAnnotation.java

```diff
@@ -64,6 +68,13 @@ public boolean allowsTopLevelDefinition() {
     return allowTopLevelDefinition;
   }
 
+  /**
+   * @return any subtitutionGroup the element might extend from
+   */
+  public SubstitutionGroup getSubstitutionGroup() {
```

Comment:
what if this is an optional? the hints saying that this has an "empty" substitution group is a bit misleading, better to say no custom group was defined and return optional empty if the paremeterized substitutionGroup was empty.